test/proposal_test.rb

```diff
@@ -21,326 +21,74 @@
+      allow(Y2Storage::Proposal::VolumesGenerator).to receive(:new).and_return volumes_generator
+      settings.root_device = "/dev/sda"
     end
 
-    context "in a windows/linux multiboot PC" do
-      let(:scenario) { "windows-linux-multiboot-pc" }
-
-      context "using LVM" do
-        let(:lvm) { true }
-
-        context "with a separate home" do
-          let(:separate_home) { true }
-          include_examples "proposed layout"
-        end
-
-        context "without separate home" do
-          let(:separate_home) { false }
-          include_examples "proposed layout"
-        end
-      end
-
-      context "not using LVM" do
-        let(:lvm) { false }
-
-        context "with a separate home" do
-          let(:separate_home) { true }
-          include_examples "proposed layout"
-        end
-
-        context "without separate home" do
-          let(:separate_home) { false }
-          include_examples "proposed layout"
-        end
-      end
+    let(:scenario) { "swaps" }
+    let(:windows_partitions) { {} }
+    let(:volumes_generator) do
+      base_volumes = [
+        planned_vol(mount_point: "/", type: :ext4, desired: 500.MiB, max: 500.MiB)
+      ]
+      all_volumes = [
+        planned_vol(mount_point: "/", type: :ext4, desired: 500.MiB, max: 500.MiB),
+        planned_vol(mount_point: "swap", reuse: "/dev/sda3"),
+        planned_vol(mount_point: "swap", type: :swap, desired: 500.MiB, max: 500.MiB),
+        planned_vol(mount_point: "swap", type: :swap, desired: 500.MiB, max: 500.MiB),
+        planned_vol(mount_point: "swap", type: :swap, desired: 500.MiB, max: 500.MiB)
+      ]
+      instance_double(
+        "Y2Storage::Proposal::VolumesGenerator",
+        base_volumes: Y2Storage::PlannedVolumesList.new(base_volumes),
+        all_volumes:  Y2Storage::PlannedVolumesList.new(all_volumes)
+      )
     end
 
-    context "in a linux multiboot PC" do
-      let(:scenario) { "multi-linux-pc" }
-      let(:windows_partitions) { {} }
-
-      context "using LVM" do
-        let(:lvm) { true }
-
-        context "with a separate home" do
-          let(:separate_home) { true }
-          include_examples "proposed layout"
-        end
-
-        context "without separate home" do
-          let(:separate_home) { false }
-          include_examples "proposed layout"
-        end
-      end
-
-      context "not using LVM" do
-        let(:lvm) { false }
-
-        context "with a separate home" do
-          let(:separate_home) { true }
-          include_examples "proposed layout"
-        end
-
-        context "without separate home" do
-          let(:separate_home) { false }
-          include_examples "proposed layout"
-        end
-      end
+    it "reuses suitable swap partitions" do
+      proposal.propose
+      sda3 = proposal.devices.partitions.with(name: "/dev/sda3").first
+      expect(sda3).to match_fields(
+        mountpoint: "swap",
+        uuid:       "33333333-3333-3333-3333-33333333",
```

Comment:
I assume you made up that UUID: In general UUIDs follow versions and rules, see https://en.wikipedia.org/wiki/Universally_unique_identifier. The one here might be invalid.